src/components/fields/DataSelector.js

```diff
@@ -44,7 +44,17 @@ class DataSelector extends Component {
     }
     const update = {};
     if (this.dataSrcExists) {
-      update[this.srcAttr] = value;
+      const srcAttr = this.srcAttr;
+      update[srcAttr] = value;
+
+      if (!value) {
+        /*
+         * when !value, then we're deleting ${attr}src, let's also clan up the
+         * related src to prevent issues when switching plots (i.e. candlestick -> box plot)
+         * ex: clean up both xsrc and x from gd.data
+         */
+        update[srcAttr.replace('src', '')] = value;
```

Comment:
should this be just `update[this.props.attr] = null` ?

Reply:
or maybe merge this into the next if-block and always dereference it, even if it's null. that way if value is falsey then we get undefined back from dataSources.